src/Microsoft.OData.Core/ODataContextUrlInfo.cs

```diff
@@ -294,19 +294,14 @@ private static string ComputeNavigationPath(EdmNavigationSourceKind kind, ODataU
                 return null;
             }
 
-            string navigationPath = null;
-            if (kind == EdmNavigationSourceKind.ContainedEntitySet && odataUri != null && odataUri.Path != null)
+            if (odataUri == null)

```

Comment:
From meeting with @mikepizzo:

```
public void Compute(ODataUri uri)
{
	Internal(uri);
}

public void Compute(Slim slim)
{
	Internal(new ODataUri(slim.ODataUri));
}

private void Internal(ODataUri)
{
}
```

Reply:
Create an 8.0 work item for rationalizing the two types

Reply:
I created [this](https://github.com/OData/odata.net/issues/2696) work item to track the 8.0 work.

@habbes do you have any thoughts on if `ComputeNavigationPathImpl` should take in an `ODataUri` vs `ODataUriSlim`?

Reply:
There are actually 3 references stored in the `ODataUriSlim`, so this is *not* as performant.

Reply:
I've added additional context to the work item and reverted this unification. Also added an additional test for the "slim" variant now that they are separate code paths again.